ext/vernier/vernier.cc

```diff
@@ -947,6 +939,13 @@ class ThreadTable {
                     if (thread.state == Thread::State::RUNNING) {
                         thread.pthread_id = pthread_self();
                         thread.native_tid = get_native_thread_id();
+                        if (thread.name == "") {
+                            // rb_inspect should be safe here, as RUNNING should correspond to RESUMED hook from internal_thread_event_cb
+                            // which is called with GVL per https://github.com/ruby/ruby/blob/v3_3_0/include/ruby/thread.h#L247-L248
+                            VALUE thread_str = rb_inspect(th);
+                            if (RTEST(thread_str))
+                                thread.name = StringValueCStr(thread_str);
+                        }
```

Comment:
I don't think this is safe here as both of these can raise and we're not ready for that (and holding onto some locks). I'll remove this before merging.

Reply:
@jhawthorn ah my bad, makes sense though. Thanks!

Reply:
All good! It's helpful actually as I was able to use this for a regression test 😊